internal/k8s/controller.go

```diff
@@ -976,8 +983,56 @@ func (lbc *LoadBalancerController) syncVirtualServer(task task) {
 				glog.Errorf("Error when updating the status for VirtualServerRoute %v/%v: %v", vsr.Namespace, vsr.Name, err)
 			}
 		}
+
+		vsrkey := fmt.Sprintf("%s/%s", vsr.Namespace, vsr.Name)
+		vsrList = removeVirtualServerRouteByKey(vsrkey, vsrList)
+	}
+	reason := "Ignored"
+	for _, vsr := range vsrList {
+		msg := fmt.Sprintf("Ignored by VirtualServer %v/%v", vs.Namespace, vs.Name)
+		lbc.recorder.Eventf(vsr, api_v1.EventTypeWarning, "Ignored", msg)
+		err := lbc.statusUpdater.UpdateVirtualServerRouteStatus(vsr, conf_v1.StateInvalid, reason, msg)
+		if err != nil {
+			glog.Errorf("Error when updating the status for VirtualServerRoute %v/%v: %v", vsr.Namespace, vsr.Name, err)
+		}
 	}
+}
 
+func removeVirtualServerRouteByKey(vsrkey string, vsrList []*conf_v1.VirtualServerRoute) []*conf_v1.VirtualServerRoute {
+	for i, r := range vsrList {
+		key := fmt.Sprintf("%s/%s", r.Namespace, r.Name)
+		if key == vsrkey {
+			if len(vsrList) > 1 {
+				vsrList = append(vsrList[:i], vsrList[i+1:]...)
+			} else {
+				vsrList = []*conf_v1.VirtualServerRoute{}
+			}
+		}
+	}
+
+	return vsrList
+}
+
+func findAllVirtualServerRoutesForVirtualServer(key string, vsrs []*conf_v1.VirtualServerRoute) []*conf_v1.VirtualServerRoute {
+	var result []*conf_v1.VirtualServerRoute
+	for _, r := range vsrs {
+		if r.Status.ReferencedBy == key {
```

Comment:
The status of the resource is not a reliable source, unfortunately. for one thing, status reporting could be disabled in the IC. Additionally, there are periods when status isn't reported even if it is enabled (when a leader is being elected).

In the `Configurator`, we already track which VSs and VSRs are currently present in the configuration. Note the map `virtualServers      map[string]*VirtualServerEx`. This means you can get the list of VSRs for a VS.

Now for the case when you delete the VS configuration (either on VS deletion or misconfiguration), before deleting the VSR, you can the list of its VSRs from Configurator. For example, you can add a method `vsrs := lbc.cnf.GetVirtualServerRoutesForVirtualServer(key)`. After the deletion, you can just go through the list and report events and update statuses of those VSRs.

Now for the case when a VS stops referencing some VSR, you can do the following:
* before you call `AddOrUpdateVirtualServer`, call `previousVSRs := lbc.cnf.GetVirtualServerRoutesForVirtualServer(key)` to get the old list of VSRs.
* After the call, compare that list with the  current list (`vsEx.VirtualServerRoutes`). For the ones from `previousVSRs` that are (1) not present in `vsEx.VirtualServerRoutes` and (2) not present in `vsrErrors`, we should report `NoVirtualServersFound` event and set the status Invalid. 

Let me know what you think